cpplint/cpplint.py

```diff
@@ -1776,13 +1776,37 @@ def GetHeaderGuardCPPVariable(filename):
 
   fileinfo = FileInfo(filename)
   file_path_from_root = fileinfo.RepositoryName()
-  if _root:
+
+  def FixupPathFromRoot():
+    # Process the file path with the --root flag if it was set.
+    if not _root:
+      return file_path_from_root
+
     suffix = os.sep
     # On Windows using directory separator will leave us with
     # "bogus escape error" unless we properly escape regex.
     if suffix == '\\':
       suffix += '\\'
-    file_path_from_root = re.sub('^' + _root + suffix, '', file_path_from_root)
+
+    # root behavior:
+    #   --root=subdir , lstrips subdir from the header guard
+    (maybe_path, subs_made) = re.subn('^' + _root + suffix, '', file_path_from_root)
```

Comment:
Please don’t do this, it’s dangerous. Take a look at re.escape(). That’ll probably eliminate the need for the “+= '\\'” a couple of lines above.